sonic_y_cable/y_cable_base.py

```
@@ -0,0 +1,1157 @@
+
+    def get_anlt(self, target):
+        """
+        This API gets the auto-negotiation + link training (AN/LT) mode of the cable for corresponding port.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the actual target to get the AN/LT on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+        Returns:
+            a boolean, True if auto-negotiation + link training (AN/LT) is enabled
+                     , False if auto-negotiation + link training (AN/LT) is not enabled
+        """
+
+        raise NotImplementedError
+
+    def get_event_log(self, clear_on_read=False):
+        """
+        This API returns the event log of the cable
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            clear_on_read:
+                a boolean, True if the log has to be cleared after read
+                         , False if the log is not to be cleared after read
+
+        Returns:
+           list:
+              a list of strings which correspond to the event logs of the cable
+        """
+
+        raise NotImplementedError
+
+    def get_pcs_stats(self, target):
+        """
+        This API returns the pcs statistics of the cable
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the actual target to get the stats on:
+                     TARGET_NIC -> NIC,
+                     TARGET_TOR_A -> TORA,
+                     TARGET_TOR_B -> TORB
+
+        Returns:
+           a dictionary:
+               a detailed format agreed upon by vendors
+        """
+
+        raise NotImplementedError
+
+    def get_fec_stats(self, target):
```

```suggestion
    def get_fec_stats(self, target, clear_on_read=False):
```
Is it necessary to add clear_on_read to reset the fec statistics?

---

if the feature cannot be supported, you can return an error when the caller tries to enable clear_on_read= True;
I believe other vendors might be able to support this, hence it is kept as an argument.